### DISCUSSION TITLE
Absorb TeslaPowerDash?

### DISCUSSION THREAD

**Discussion by Nakatomi2010** [teslamate-org/teslamate#1195] [Ideas]
https://github.com/rhodesman/teslaPowerDash

This project has been a little dead, and the Middleware piece is a bit dated.  Would be nice to see TeslaMate mate with more Tesla products.

TeslaPowerDash already uses the Tesla API and Grafana, should mostly just be a matter of integrating into the database and such.

**Comment by natrlhy** [comment]
I'd love to see this, too. I now have a Tesla car, solar panels and 2 powerwalls and would love to help in any way.

**Reply by Nakatomi2010** [reply]
Same boat I'm in, but two Teslas an one Powerwall.

I mean, the TeslaPowerDash thing has almost everything.  The only obnoxious piece about it is that you can't get "real time" data from the panels without the "Middleware" bullcrap, which basically just interfaces with the Powerwall itself.  The historical graphs come from the Tesla API, and get dumped directly to the database for Grafana to pull.

Historical data shouldn't be hard to populate, but live data is going to be the tricky bit.

**Reply by natrlhy** [reply]
Agree, for me, I've wanted a central way/location to ingest the historical data of /grid/solar/pw/home usage, graph things and calculate things, etc. Real-time data does seem to be the tricky part as that all seems to be coming from the Tesla Energy Gateway (https://teg locally)

**Comment by blharveymi** [comment]
any update on this or adding the information that jseter started with other solar idea?
